Sources/SotoCore/AWSClient+Paginate+async.swift

```diff
@@ -0,0 +1,122 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Soto for AWS open source project
+//
+// Copyright (c) 2017-2020 the Soto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Soto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if compiler(>=5.5) && $AsyncAwait
```

Comment:
This is nice. How/where is `$AsyncAwait` defined?

Reply:
I just copied this from swift nio. Not sure where $AsyncAwait comes from exactly.